src/core/Main.pm

```diff
@@ -150,7 +150,37 @@ my sub MAIN_HELPER($retval = 0) is hidden_from_backtrace {
     @matching_candidates .=grep: {!has-unexpected-named-arguments($_.signature, $n)};
     # If there are still some candidates left, try to dispatch to MAIN
     if +@matching_candidates {
-        return $m(|@($p), |%($n));
+        # verify that we don't dispatch multiple occurences of a command line argument
+        # to a parameter that is a Scalar
```

Comment:
My thought would be that we can still dispatch to them, but should tie-break. So `sub MAIN(:$arg) {}` could still receive `program --arg=1 --arg=2`, but `multi MAIN(:@arg) {}` beats `multi MAIN(:$arg) {}` with that invocation… OTOH, invocations like `-n=1,2` are still NYI, which in my mind is different than `-n=1 -n=2`, and there doesn't seem to be spec on the latter.